### ISSUE TITLE
Missing required module 'COpenCombineHelpers'

### ISSUE THREAD

**Issue by fotiDim** [broadwaylamb/OpenCombine#94]
The moment I do "import OpenCombine" I am getting this error `Missing required module 'COpenCombineHelpers'`. Is there something additional to be done?

**Comment by broadwaylamb** [comment]
Hello! No, everything should just work.

What is your setup? Are you using the library as a Swift Package Manager dependency, and what version?

I'd really like to help you with this, so any details about how to reproduce the issue would be much appreciated!

**Comment by fotiDim** [comment]
Yes Swift Package Manager. I tried 0.4 and 0.5. I got around the error by adding "OpenCombine" under "Frameworks and Libraries". However now I ran into the next error which is `Unknown attribute 'Published'` when I do `@Published private var searchText: String = ""`

**Comment by broadwaylamb** [comment]
> I got around the error by adding "OpenCombine" under "Frameworks and Libraries".

Oh, Published.

Published doesn't work yet, unfortunately. Believe it or not, I'm coding it right at this moment, so expect it to be ready in the next version.

The error you get, however, can probably by using `@OpenCombine.Published`. Why? Because Foundation re-exports `Published` from Combine, so there can be a name collision if both Foundation and OpenCombine are imported.

**Comment by broadwaylamb** [comment]
> I got around the error by adding "OpenCombine" under "Frameworks and Libraries".

Well, yes, that needs to be done. When you add a Swift package as a dependency, Xcode suggests you a list of the package targets with checkboxes. Activating those checkboxes basically adds the target to "Frameworks and Libraries":

<img width="851" alt="Screenshot 2019-10-26 at 13 58 24" src="https://user-images.githubusercontent.com/16309982/67618468-bd379f80-f7f8-11e9-917f-e76e878a1aee.png">


> However now I ran into the next error which is `Unknown attribute 'Published'` when I do `@Published private var searchText: String = ""`

Oh, Published.

Published doesn't work yet, unfortunately. Believe it or not, I'm coding it right at this moment, so expect it to be ready in the next version.

The error you get, however, can probably be fixed by writing `@OpenCombine.Published`. Why? Because Foundation re-exports `Published` from Combine, so there can be a name collision if both Foundation and OpenCombine are imported.

**Comment by fotiDim** [comment]
I see. It is a good idea to put that in the Readme. I had no idea what OpenCombineDispatch is and if it was needed.

I see. Regarding iOS 13+ does the library fall back to Apple's Combine or still uses the OpenCombine implementation? Would something like that work?
```
#if canImport(Combine)
import Combine
#else
import OpenCombine
#endif
```

**Comment by broadwaylamb** [comment]
>It is a good idea to put that in the Readme. I had no idea what OpenCombineDispatch is and if it was needed.

I'll definitely do. OpenCombineDispatch extends the `DispatchQueue` class and adds conformance to the `Scheduler` protocol. It is extracted into a different target because OpenCombine (just like Combine) doesn't depend on Dispatch, or Foundation, or anything like that. So if you don't need that conformance, you are free to exclude that target. But you probably do.

> Regarding iOS 13+ does the library fall back to Apple's Combine or still uses the OpenCombine implementation? Would something like that work?

Frankly, I'm not sure it is possible to fallback to Combine on iOS 13+. The reason for that is that would require *runtime* checks like `if #available(iOS 13.0)`, rather then compile-time checks like in your example. And those runtime checks would require a whole lot of complexity and would likely cause performance regressions.

So, answering to your question, even on iOS 13+ OpenCombine implementation is used, and there's not really anything we can do about it right now :(
